calyx-py/calyx/builder.py

```diff
@@ -1273,14 +1352,18 @@ def infer_width(self, port_name) -> int:
         ):
             if port_name in ("left", "right"):
                 return inst.args[0]
-        if prim in ("comb_mem_d1", "seq_mem_d1"):
+        if prim in ("comb_mem_d1", "seq_mem_d1", "comb_mem_d2", "seq_mem_d2"):
             if port_name == "write_en":
                 return 1
+            if "d2" in prim and port_name == "addr0":
+                return inst.args[3]
+            if "d2" in prim and port_name == "addr1":
+                return inst.args[4]
             if port_name == "addr0":
                 return inst.args[2]
             if port_name == "in":
                 return inst.args[0]
-            if prim == "seq_mem_d1" and port_name == "content_en":
+            if "seq_mem" in prim and port_name == "content_en":
```

Comment:
Super! More width-inference has become my little pet project haha, so thank you for pushing the change through